### PR TITLE
Preparing to localize strings in ArtifactEngine library.

### PR DESCRIPTION
**Pull request by kasubram**

### CODE REVIEW

File: Extensions/ArtifactEngine/Providers/filesystemProvider.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export class FilesystemProvider implements models.IArtifactProvider {
             var items: models.ArtifactItem[] = [];
             fs.readdir(itemsPath, (error, files) => {
                 if (!!error) {
-                    Logger.logMessage("Unable to read directory " + itemsPath + ". Error: " + error);
+                    Logger.logError(tl.loc("UnableToReadDirectory", itemsPath, error));
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/Providers/filesystemProvider.ts]: logging error will end up failing task. @omeshp is this fine?

File: Extensions/ArtifactEngine/Engine/artifactEngine.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ import { TicketState } from '../Models/ticketState';
 export class ArtifactEngine {
     processItems(sourceProvider: models.IArtifactProvider, destProvider: models.IArtifactProvider, artifactEngineOptions?: ArtifactEngineOptions): Promise<models.ArtifactDownloadTicket[]> {
         var artifactDownloadTicketsPromise = new Promise<models.ArtifactDownloadTicket[]>((resolve, reject) => {
+            tl.setResourcePath(path.join(path.dirname(__dirname), 'lib.json'));
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/Engine/artifactEngine.ts]: lib.json seems generic name. will it cause conflicts?

File: Extensions/ArtifactEngine/Providers/webProvider.ts [hunk 3]
```diff
@@ -103,7 +104,7 @@ export class WebProvider implements models.IArtifactProvider {
 
                             resolve(items);
                         } catch (error) {
-                            Logger.logMessage("Failed to parse response body: " + body + " , got error : " + error);
+                            Logger.logError(tl.loc("FailedToParseResponse", body, error));
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/Providers/webProvider.ts]: @omeshp - should this be logMessage only?

File: Extensions/ArtifactEngine/lib.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/lib.json]: rename to strings.json or messages.json?

File: Extensions/ArtifactEngine/package.json [hunk 5]
```diff
@@ -18,7 +18,7 @@
   "dependencies": {
     "handlebars": "4.0.10",
     "minimatch": "3.0.2",
-    "vsts-task-lib": "^2.1.0",
+    "vsts-task-lib": "2.2.1",
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/package.json]: this require 2.2.1 version of task lib?

File: externals.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
     "npm": {
-        "vsts-task-lib": "0.7.4"
+        "vsts-task-lib": "2.2.0"
```
Comment (GitHubSriramB) [inline externals.json]: @omeshp - do you see any concerns with bumping up vsts-task-lib so close to on-prem release?

File: Extensions/ArtifactEngine/package.json (hunk 5 above)
Comment (kasubram) [reply Extensions/ArtifactEngine/package.json]: Yes we require 2.2.1, this allows to set multiple resource file for the task.

File: Extensions/ArtifactEngine/Providers/filesystemProvider.ts (hunk 1 above)
Comment (omeshp) [reply Extensions/ArtifactEngine/Providers/filesystemProvider.ts]: we should change this to logMessage.

File: gulpfile.js [hunk 7]
```diff
@@ -289,6 +561,10 @@ function compileUIExtensions(extensionRoot) {
 gulp.task("build", ["compileNode"], function() {
 });
 
+gulp.task("handoff", function(cb) {
```
Comment (omeshp) [inline gulpfile.js]: update read me to include this.

File: Extensions/ArtifactEngine/Engine/artifactEngine.ts (hunk 2 above)
Comment (kasubram) [reply Extensions/ArtifactEngine/Engine/artifactEngine.ts]: No, it wont cause any conflicts..

**Pull request merged**